### PR TITLE
Fix latest deps CI

### LINKED ISSUE #13731: CI run against latest deps is failing
See https://github.com/matrix-org/synapse/actions/runs/3005704528

### PR DESCRIPTION
**Pull request by erikjohnston**
Hopefully fixes #13731 and #13732.

Broke by #12595

### CODE REVIEW

File: .github/workflows/latest_deps.yml [hunk 1]
```diff
@@ -26,12 +26,19 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+            toolchain: stable
```
Comment (DMRobertson) [inline .github/workflows/latest_deps.yml:32]: For latest deps, do we want stable or nightly? Stable is probably the closest to what we do for Python packages.

File: .github/workflows/latest_deps.yml [hunk 2]
```diff
@@ -26,12 +26,19 @@ jobs:
...
+            override: true
+      - uses: Swatinem/rust-cache@v2
+
       # The dev dependencies aren't exposed in the wheel metadata (at least with current
       # poetry-core versions), so we install with poetry.
       - uses: matrix-org/setup-python-poetry@v1
         with:
           python-version: "3.x"
-          poetry-version: "1.2.0b1"
+          poetry-version: "1.2.0"
```
[the remark this answers is not in the document]
Comment (DMRobertson) [reply .github/workflows/latest_deps.yml:41]: Last time I checked, GHA didn't let you use anchors and merges

**Review by DMRobertson — APPROVED** [review]
SGTM, but probably worth manually triggering the two jobs after this merges to confirm they work. (Unless you can run those jobs against this PR directly?)

File: .github/workflows/latest_deps.yml (hunk 1 above)
Comment (erikjohnston) [reply .github/workflows/latest_deps.yml:32]: Yeah, we could also do beta I guess? I don't think it matters much

Comment (DMRobertson) [reply .github/workflows/latest_deps.yml:32]: Sure. (Just thinking out loud here.)

File: .github/workflows/latest_deps.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (DMRobertson) [reply .github/workflows/latest_deps.yml:41]: Refs: https://github.com/actions/runner/issues/1182 and https://github.com/community/community/discussions/4501

**Pull request merged**